core/commands/follow.go

```diff
@@ -0,0 +1,182 @@
+package commands
+
+import (
+	"errors"
+	"io"
+	"strings"
+
+	cmds "github.com/ipfs/go-ipfs/commands"
+	e "github.com/ipfs/go-ipfs/core/commands/e"
+
+	"gx/ipfs/QmUyfy4QSr3NXym4etEiRyxBLqqAeKHJuRdi8AACxg63fZ/go-ipfs-cmdkit"
+)
+
+type ipnsFollowResult struct {
+	OK bool
+}
+
+var IpnsFollowCmd = &cmds.Command{
+	Helptext: cmdkit.HelpText{
+		Tagline: "Follow IPNS names.",
+		ShortDescription: `
+Periodically resolve and optionally pin IPNS names in the background.
+`,
+	},
+	Subcommands: map[string]*cmds.Command{
+		"add":    ipnsFollowAddCmd,
+		"pin":    ipnsFollowPinCmd,
```

Comment:
I'd collapse pin and add to `add --pin` or something like that.

Reply:
Good point.

Reply:
Done.